src/plugins/workspace/server/saved_objects/workspace_ui_settings_client_wrapper.ts

```diff
@@ -0,0 +1,122 @@
+    return this.getScopedClient?.(request, {
+      includedHiddenTypes: [WORKSPACE_TYPE],
+      excludedWrappers: [WORKSPACE_UI_SETTINGS_CLIENT_WRAPPER_ID],
+    }) as SavedObjectsClientContract;
+  }
+
+  public setScopedClient(getScopedClient: SavedObjectsServiceStart['getScopedClient']) {
+    this.getScopedClient = getScopedClient;
+  }
+
+  public wrapperFactory: SavedObjectsClientWrapperFactory = (wrapperOptions) => {
+    const getUiSettingsWithWorkspace = async <T = unknown>(
+      type: string,
+      id: string,
+      options: SavedObjectsBaseOptions = {}
+    ): Promise<SavedObject<T>> => {
+      const { requestWorkspaceId } = getWorkspaceState(wrapperOptions.request);
+
+      /**
+       * When getting ui settings within a workspace, it will combine the workspace ui settings with
+       * the global ui settings and workspace ui settings have higher priority if the same setting
+       * was defined in both places
+       */
+      if (type === 'config' && requestWorkspaceId) {
+        const configObject = await wrapperOptions.client.get<Record<string, any>>(
+          'config',
+          id,
+          options
+        );
+
+        const workspaceObject = await this.getWorkspaceTypeEnabledClient(
+          wrapperOptions.request
+        ).get<WorkspaceAttribute>(WORKSPACE_TYPE, requestWorkspaceId);
+
+        configObject.attributes = {
+          ...configObject.attributes,
+          ...workspaceObject.attributes.uiSettings,
+        };
+
+        return configObject as SavedObject<T>;
+      }
+
+      return wrapperOptions.client.get(type, id, options);
+    };
+
+    const updateUiSettingsWithWorkspace = async <T = unknown>(
+      type: string,
+      id: string,
+      attributes: Partial<T>,
+      options: SavedObjectsUpdateOptions = {}
+    ): Promise<SavedObjectsUpdateResponse<T>> => {
+      const { requestWorkspaceId } = getWorkspaceState(wrapperOptions.request);
+
+      /**
+       * When updating ui settings within a workspace, it will update the workspace ui settings,
+       * the global ui settings will remain unchanged.
+       */
+      if (type === 'config' && requestWorkspaceId) {
+        const configObject = await wrapperOptions.client.get<Record<string, any>>(
+          'config',
+          id,
+          options
+        );
+        const savedObjectsClient = this.getWorkspaceTypeEnabledClient(wrapperOptions.request);
+
+        const workspaceObject = await savedObjectsClient.get<WorkspaceAttribute>(
+          WORKSPACE_TYPE,
+          requestWorkspaceId
+        );
+
+        const workspaceUpdateResult = await savedObjectsClient.update<WorkspaceAttribute>(
+          WORKSPACE_TYPE,
+          requestWorkspaceId,
+          {
+            ...workspaceObject.attributes,
+            uiSettings: { ...workspaceObject.attributes.uiSettings, ...attributes },
+          },
+          options
+        );
+
+        configObject.attributes = { ...workspaceUpdateResult.attributes.uiSettings };
```

Comment:
`update` function only returns the saved object with the fields updated, what I'm trying to do here is to return the global config object with the fields updated to workspace uiSettings.

This may sound a bit hacky, because the `update` function originally update the global config object, but in fact it updates the workspace object, but in the end, the update method should return a config object.

Reply:
```suggestion
        configObject.attributes = workspaceUpdateResult.attributes.uiSettings;
```
I mean can we do this?